deepcell_spots/applications/spot_detection.py

```diff
@@ -107,6 +107,44 @@ def __init__(self, model=None):
             dataset_metadata=self.dataset_metadata,
             model_metadata=self.model_metadata)
 
+    def _run_model(self,
+                   image,
+                   batch_size=4,
+                   pad_mode='constant',
+                   preprocess_kwargs={}):
+        """Run the model to generate output probabilities on the data.
+
+        Args:
+            image (numpy.array): Image with shape ``[batch, x, y, channel]``
+            batch_size (int): Number of images to predict on per batch.
+            pad_mode (str): The padding mode, one of "constant" or "reflect".
+            preprocess_kwargs (dict): Keyword arguments to pass to
+                the preprocessing function.
+
+        Returns:
+            numpy.array: Model outputs
+        """
+        # Preprocess image if function is defined
+        image = self._preprocess(image, **preprocess_kwargs)
+
+        # Tile images, raises error if the image is not 4d
+        tiles, tiles_info = self._tile_input(image, pad_mode=pad_mode)
+
+        # Run images through model
+        t = timeit.default_timer()
+        output_tiles = self.model.predict(tiles, batch_size=batch_size)
+        self.logger.debug('Model inference finished in %s s',
+                          timeit.default_timer() - t)
+
+        # Untile images
+        output_images = self._untile_output(output_tiles, tiles_info)
+
+        # restructure outputs into a dict if function provided
+        formatted_images = {name: pred for name, pred in zip(self.model.output_names,
+                                                             output_images)}
+
+        return formatted_images
+
```

Comment:
Why did you decide to duplicate _run_model with the addition of lines 142-144 instead of just adding those lines after calling _run_model in _predict?

Reply:
I replaced the line calling `_format_model_output` because it didn't have a way to pass arguments in the `_format_model_output_fn`. You're right that I could have just done this outside of _run_model but my thought was that the line I added serves the same purpose as the `_format_model_output` call.

Reply:
I would try to avoid duplicating a core function like _run_model especially if there is a way to pass in a format function. It's going to make the spots repo harder to maintain in the long run if we ever do a major refactor of the deepcell application code. If you want keep the output names exposed, you could make it a parameter in a function `format_spots_output` and then when you pass it in to the application to define `_format_model_output_fn` you could first override it with `self.model.output_names`, e.g. 
```
def _format_spots_output(output):
	return format_spots_otuput(output, output_names=self.model.output_names)
```
I'll let it go if you think its too complicated, but take a look first.

Reply:
If you think it's cleaner to move it out of `_run_model` I can change it

Reply:
I would prefer having that change.